src/env.h

```diff
@@ -608,6 +608,14 @@ class Environment {
 
   std::unordered_multimap<int, loader::ModuleWrap*> module_map;
 
+  struct PackageConfig {
+    bool exists;
+    bool is_valid;
+    bool has_main;
+    std::string main;
+  };
```

Comment:
Would still be nice if these were const.

Reply:
As far as I can tell getting this to work would mean inlining the `kEmptyPackage` constructor at each place it is assigned through an emplace call, which doesn't seem very nice.

Reply:
(as in `kEmptyPackage` would be replaced by these inline constructors?)

Reply:
`map.emplace(std::make_pair(key, kEmptyPackage))` should work, I think.  If it doesn't, you could add a `PackageConfig() = default` constructor and emplace `PackageConfig()` instead.

Reply:
The first case didn't work unfortunately. In the second case it would need to be a `PackageConfig(true)` and a `PackageConfig(false)` to avoid a full expansion of arguments. But `= default` doesn't seem to be compatible with default constructor arguments as far as I can tell?

Reply:
I managed to get the `const`s to work out, but it's pretty awful code to look at.